src/PSSparseServerInterface.cpp

```diff
@@ -34,10 +34,16 @@ PSSparseServerInterface::PSSparseServerInterface(const std::string& ip, int port
   std::memset(serv_addr.sin_zero, 0, sizeof(serv_addr.sin_zero));
 
   // Connect to the server
-  if (::connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
-    throw std::runtime_error(
-        "Client could not connect to server."
-        " Address: " + ip + " port: " + std::to_string(port));
+  int ret = -1;
```

Comment:
I don't like this change. With this change, if we create a PSSparseServerInterface that can't connect to the PS we are stuck in this loop forever.

I suggest to move this connect code part to a separate PSSparseServerInterface::connect method. This method throws an exception in case of connection error which allows the client to retry connecting.

Reply:
Makes sense. Working on implementing this.